rust-workspace/crates/ntc-asset-manager/src/drt.rs

```diff
@@ -0,0 +1,282 @@
+            data_package: [2u8; 32].to_vec(),
+            data_url: String::from("https://host2.example.com"),
+        };
+
+        let builder = DrtConfigBuilder::new(CREATOR_ADDRESS, note)
+            .name("DRT")
+            .meta_data_hash([5u8; 32].as_slice())
+            .url("https://drt.example.com")
+            .unwrap()
+            .supply(10);
+        builder.build()
+    }
+
+    fn get_suggested_txn_params() -> SuggestedTransactionParams {
+        SuggestedTransactionParams {
+            genesis_id: String::from("id"),
+            genesis_hash: HashDigest([7u8; 32]),
+            consensus_version: String::from("v2"),
+            fee_per_byte: MicroAlgos(2u64),
+            min_fee: MicroAlgos(1u64),
+            first_valid: Round(10),
+            last_valid: Round(20),
+        }
+    }
+
+    fn get_test_transaction() -> Transaction {
+        let payment = Payment {
+            sender: Address(CREATOR_ADDRESS),
+            receiver: Address([2u8; 32]),
+            amount: MicroAlgos(3u64),
+            close_remainder_to: None,
+        };
+        TxnBuilder::new(
+            algonaut::transaction::builder::TxnFee::Fixed(MicroAlgos(4u64)),
+            Round(10),
+            Round(20),
+            HashDigest([7u8; 32]),
+            algonaut::transaction::TransactionType::Payment(payment),
+        )
+        .build()
+        .unwrap()
+    }
+
+    fn signed_transaction(txn: &Transaction) -> SignedTransaction {
+        SignedTransaction {
+            transaction: txn.clone(),
+            transaction_id: String::from("id"),
+            sig: algonaut::transaction::transaction::TransactionSignature::Single(Signature(
+                [6u8; 64],
+            )),
+        }
+    }
+
+    #[tokio::test]
+    async fn submit_drt_success() {
+        let drt_config = build_drt_config();
+
+        let mut mock_account = Account::new();
+        let mut mock_algod = Algod::new();
+
+        let txn = get_test_transaction();
+        mock_account
+            .expect_address()
+            .once()
+            .return_const(Address(CREATOR_ADDRESS));
+        mock_algod
+            .expect_suggested_transaction_params()
+            .once()
+            .return_once(|| Ok(get_suggested_txn_params()));
+
+        let signed_txn = signed_transaction(&txn);
+        mock_account
+            .expect_sign_transaction()
+            .once()
+            .return_once(|_| Ok(signed_txn));
+
+        mock_algod
+            .expect_broadcast_signed_transaction()
+            .once()
+            .withf(|t| *t == signed_transaction(&get_test_transaction()))
+            .return_once(|_| {
+                let tx_id = String::from("id");
+                Ok(TransactionResponse { tx_id })
+            });
+
+        let response = drt_config.submit(&mock_algod, &mock_account).await.unwrap();
+        assert!(matches!(response, TransactionResponse { .. }));
```

Comment:
Cool things learnt  about unit tests:
Can run the following to output `println!` statements in test output and specificy `--lib` to just run the unit tests - https://doc.rust-lang.org/cargo/commands/cargo-test.html
```
cargo test --lib -- --nocapture
```

Can also make the assert more specific with but the above is sufficient:
```
assert_eq!(response.tx_id, "id");
```